src/main/java/com/linecorp/armeria/server/metrics/Status.java

```diff
@@ -0,0 +1,25 @@
+package com.linecorp.armeria.server.metrics;
+
+import com.linecorp.armeria.common.SessionProtocol;
+
+/**
+ * Server response status for {@link ServiceMetricConsumer}
+ */
+final class Status {
+    private final SessionProtocol sessionProtocol;
+    private final int code;
+
+    private Status(SessionProtocol sessionProtocol, int code) {
+        this.sessionProtocol = sessionProtocol;
+        this.code = code;
+    }
+
+    public static Status of(SessionProtocol protocol, int code) {
+        return new Status(protocol, code);
```

Comment:
As long as decorator uses http statuscode at its internal, I think primitive int seems enough. let me implement `int code`, and please give feedback if other concerns.